sqlanalytics/resource_widget_test.go

```diff
@@ -500,6 +500,40 @@ func TestWidgetCreateWithPositionAndAutoheight(t *testing.T) {
 	assert.Equal(t, true, d.Get("position.0.auto_height"))
 }
 
+func TestWidgetReadNotFound(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "GET",
+				Resource: "/api/2.0/preview/sql/dashboards/some-uuid",
+				Response: api.Dashboard{
+					ID: "some-uuid",
+					Widgets: []json.RawMessage{
+						json.RawMessage(`
+							{
+								"id": "12345",
+								"text": "text"
+							}
+						`),
+					},
+				},
+			},
+		},
+		Resource:    ResourceWidget(),
+		Read:        true,
+		Removed:     true,
+		RequiresNew: true,
+		ID:          "some-uuid/12344",
+		InstanceState: map[string]string{
+			"dashboard_id":     "some-uuid",
+			"visualization_id": "678",
+		},
+	}.Apply(t)
```

Comment:
and for these simple tests that don't have to check for `ResourceData`, please just use the `.ApplyNoError(t)` instead of `.Apply(t)`